### PR TITLE
added remove all children from a node

### PR DESCRIPTION
**Pull request by AlexHolly**
#8125

### CODE REVIEW

File: scene/main/node.cpp [hunk 1]
```diff
@@ -2775,6 +2786,7 @@ void Node::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_children"), &Node::_get_children);
 	ClassDB::bind_method(D_METHOD("get_child:Node", "idx"), &Node::get_child);
 	ClassDB::bind_method(D_METHOD("has_node", "path"), &Node::has_node);
+	ClassDB::bind_method(D_METHOD("remove_all_children:Node"), &Node::remove_all_children);
```
Comment (karroffel) [inline scene/main/node.cpp]: `remove_all_children` doesn't return `Node`, please remove the `:Node` part, it's wrong.

**Review by karroffel — CHANGES_REQUESTED** [review]
clang-format isn't happy. Fix your formatting. Take a look at the travis page for that.

**Comment by neikeq** [comment]
`remove_child` does not `queue_free` the child, this will lead to confusion. This new method should be named `queue_free_children`, not `remove_all_children`.
I don't think there is any point in adding such method though, IMO. Doing this is already pretty simple.

**Comment by volzhs** [comment]
@neikeq I second that it should be `queue_free_children`

**Comment by akien-mga** [comment]
>  I don't think there is any point in adding such method though, IMO. Doing this is already pretty simple.

Agreed. IMO `for child in get_children_count(): child.queue_free()` is simple enough and more explicit.

**Comment by volzhs** [comment]
And I'm worrying this can cause unexpected behavior like #7142

**Comment by Shadowblitz16** [comment]
@akien-mga  please reopen this.
your logic is flawed due to gdscript being so slow

**Comment by akien-mga** [comment]
Make a benchmark before calling my logic flawed.

**Comment by Calandiel** [comment]
Well, for languages other than gdscript (C# or Rust, for example), this will very clearly be slower than the proposed API.

You need `n` calls into engine routines for `n` children (instead of just one) and you perform an allocation to get the list of children.

What kind of benchmark would you like to see?

**Pull request closed without merging**